core/src/main/java/org/apache/carbondata/core/scan/executor/impl/AbstractQueryExecutor.java

```diff
@@ -544,12 +543,7 @@ private int getKeySize(List<QueryDimension> queryDimension,
     }
     UnsafeMemoryManager.INSTANCE.freeMemoryAll(ThreadLocalTaskInfo.getCarbonTaskInfo().getTaskId());
     if (null != queryProperties.executorService) {
-      queryProperties.executorService.shutdown();
-      try {
-        queryProperties.executorService.awaitTermination(1, TimeUnit.HOURS);
-      } catch (InterruptedException e) {
-        throw new QueryExecutionException(e);
-      }
+      queryProperties.executorService.shutdownNow();
```

Comment:
Please add a comment shutdownnow is called so in case of limit query when number of limit records is already found so executors must stop all the running execution otherwise it will keep running and will hit the query performance.